src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -548,7 +555,8 @@ void ResolveContext::DispatchSuccess()
 
     for (auto & interface : interfaces)
     {
-        if (TryReportingResultsForInterfaceIndex(interface.first.first, interface.first.second))
+        auto interfaceId = interface.first.first;
+        if (TryReportingResultsForInterfaceIndex(interfaceId))
```

Comment:
The reason I couldn't add the domain here is because there are hardcoded interfaces to prioritize that do not have a domain and I have no idea what domains to use here. we need to discuss this more for me to fix this properly.

Reply:
My point is that the "prioritize" bits indeed can only pass in the interface id (and then walk through all the matches and call the two-arg version for each one), but the bit here should pass in both pieces.